Alina Nor Home Task - Unit 6

AlinaNor opened the pull request:
```
Improve the ProductFormPage - add “inactive” flag with default value is false (soft delete)
Improve the products selector for ProductListPage - ignore products with inactive flag = true Extra: Implement limit of photos
```

AlinaNor committed:
Alina Nor Home Task - Unit 6

    Improve the ProductFormPage - add “inactive” flag with default value is false (soft delete)
    Improve the products selector for ProductListPage - ignore products with inactive flag = true Extra: Implement limit of photos

barbarosh commented on client/modules/Product/pages/ProductFormPage/ProductFormPage.js:
```diff
@@ -32,14 +33,19 @@ class ProductFormPage extends Component {
-    this.state = { colors: {'color_1': {name: 'red', files: []}, 'color_2': {name: '#ffffff', files: []}},
-                   categories: this.props.categories.map(category => {return {value: category.cuid, label: category.name}})};
+    this.state = props.product || { colors: {'color_1': {name: 'red', files: []}, 'color_2': {name: '#ffffff', files: []}}, photos: [], active: false};
+    this.state.categories = this.props.categories.map(category => {return {value: category.cuid, label: category.name}});
```
Можно и так написать  this.state.categories = this.props.categories.map(category =>  {value: category.cuid, label: category.name});

AlinaNor replied:
Переделала

barbarosh commented on client/modules/Product/pages/ProductFormPage/ProductFormPage.js:
```diff
@@ -32,14 +33,19 @@ class ProductFormPage extends Component {
-    this.state = { colors: {'color_1': {name: 'red', files: []}, 'color_2': {name: '#ffffff', files: []}},
-                   categories: this.props.categories.map(category => {return {value: category.cuid, label: category.name}})};
+    this.state = props.product || { colors: {'color_1': {name: 'red', files: []}, 'color_2': {name: '#ffffff', files: []}}, photos: [], active: false};
+    this.state.categories = this.props.categories.map(category => {return {value: category.cuid, label: category.name}});
```
Тут такой код не стоит писать, сейчас это работает потому что категории будут загружены в любом случае. Такой код лучше написать в коллбеке componentWillReceiveProps и componentDidMount

AlinaNor replied:
@barbarosh Переделала

barbarosh commented on client/modules/Product/pages/ProductFormPage/ProductFormPage.js:
```diff
@@ -202,8 +234,10 @@ ProductFormPage.propTypes = {
 
-function mapStateToProps(store) {
-  return { categories: getAllCategories(store)};
+function mapStateToProps(store, props = null) {
```
Зачем тут присвоение null ?

AlinaNor replied:
@barbarosh 
Потому что если не задать, то когда создаешь новый продукт ругается что props undefined

barbarosh replied:
Точно на пропс ругается? Может на парамс ?

AlinaNor committed:
Alina Nor Home Task - Unit 6

    Moved creating of categories in state from constructor to componentDidMount and componentWillReceiveProps.

barbarosh commented on client/modules/Product/pages/ProductFormPage/ProductFormPage.js:
```diff
@@ -32,14 +33,25 @@ class ProductFormPage extends Component {
+  }
+
+  componentDidMount(){
+    this.setState({categories: this.props.categories.map(category => ({value: category.cuid, label: category.name}))});
```
Теперь дублирование кода )

AlinaNor committed:
Alina Nor Home Task - Unit 6

    Moved creating of categories in state from constructor to componentDidMount and componentWillReceiveProps.
